api/reviews_test.go

```diff
@@ -0,0 +1,185 @@
+		Convey("When the book does not exist", func() {
+			mockDataStore := &mock.DataStoreMock{
+				GetBookFunc: func(ctx context.Context, id string) (*models.Book, error) {
+					return nil, apierrors.ErrBookNotFound
+				},
+			}
+
+			ctx := context.Background()
+
+			api := Setup(ctx, host, mux.NewRouter(), mockDataStore, &hcMock)
+			response := httptest.NewRecorder()
+
+			request, err := http.NewRequest(http.MethodGet, "/books/"+bookID+"/reviews/"+reviewID, nil)
+			So(err, ShouldBeNil)
+
+			api.router.ServeHTTP(response, request)
+			Convey("Then the HTTP response code is 404", func() {
+				So(response.Code, ShouldEqual, http.StatusNotFound)
+			})
+			Convey("And the GetReview function is not called", func() {
+				So(mockDataStore.GetBookCalls(), ShouldHaveLength, 1)
+				So(mockDataStore.GetReviewCalls(), ShouldHaveLength, 0)
+			})
+		})
+
+		Convey("When the database returns a generic error", func() {
+			Convey("When the book do the database returns an unexpected error", func() {
+				mockDataStore := &mock.DataStoreMock{
+					GetBookFunc: func(ctx context.Context, id string) (*models.Book, error) {
+						return nil, errMongoDB
+					},
+				}
+
+				ctx := context.Background()
+
+				api := Setup(ctx, host, mux.NewRouter(), mockDataStore, &hcMock)
+				response := httptest.NewRecorder()
+
+				request, err := http.NewRequest(http.MethodGet, "/books/"+bookID+"/reviews/"+reviewID, nil)
+				So(err, ShouldBeNil)
+
+				api.router.ServeHTTP(response, request)
+				Convey("Then 500 InternalServerError status code is returned", func() {
+					So(response.Code, ShouldEqual, http.StatusInternalServerError)
+				})
+			})
+		})
+	})
+}
+
+func TestReviews(t *testing.T) {
+	hcMock := mock.HealthCheckerMock{}
+
+	Convey("Given an HTTP GET request to the /books/{id}/reviews/{review_id} endpoint", t, func() {
+
+		api := &API{
+			host:      host,
+			router:    mux.NewRouter(),
+			dataStore: &mock.DataStoreMock{},
+			hc:        &hcMock,
+		}
+
+		Convey("When the {review_id} is empty", func() {
+			request, err := http.NewRequest("GET", "/books/"+bookID+"/reviews/"+emptyID, nil)
+			So(err, ShouldBeNil)
+
+			response := httptest.NewRecorder()
+
+			api.getReview(response, request)
+			Convey("Then the HTTP response code is 400", func() {
+				So(response.Code, ShouldEqual, http.StatusBadRequest)
+			})
+		})
+	})
+
+	Convey("Given an HTTP GET request to the /books/{id}/reviews/{review_id} endpoint", t, func() {
+
+		api := &API{
+			host:      host,
+			router:    mux.NewRouter(),
+			dataStore: &mock.DataStoreMock{},
+			hc:        &hcMock,
+		}
+
+		Convey("When the {id} is empty", func() {
+			request, err := http.NewRequest("GET", "/books/"+emptyID+"/reviews/"+reviewID, nil)
+			So(err, ShouldBeNil)
+
+			response := httptest.NewRecorder()
+
+			api.getReview(response, request)
+			Convey("Then the HTTP response code is 400", func() {
+				So(response.Code, ShouldEqual, http.StatusBadRequest)
```

Comment:
Same as above, missing assertion to check response body for error message.

Reply:
The same type of assertion can be applied to each of the previous test scenarios in `TestReviewEndpoint`